core/src/test/java/co/cask/wrangler/executor/BasicStatisticsTest.java

```diff
@@ -149,5 +153,76 @@ public void testLargeFile() throws Exception {
       }
     }
   }
+
+  /**
+   * Test type detection using data from a 1000 line Mockaroo generated csv file
+   * @throws Exception
+   */
+  @Test
+  public void testResourceFile() throws Exception {
+    String[] directives = new String[] {
+            "parse-as-csv body , true",
+            "drop body"
+    };
+
+    List<Record> records = new ArrayList<>();
+
+    try {
+      BufferedReader bReader = new BufferedReader(new InputStreamReader(
+              this.getClass().getResourceAsStream("/" + DATA_FILE_2)));
+      String line;
+      while ((line = bReader.readLine()) != null) {
+        records.add(new Record("body", line));
+      }
+    } catch (FileNotFoundException e) {
+      e.printStackTrace();
+    }catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    Pipeline pipeline = new PipelineExecutor();
+    pipeline.configure(new TextDirectives(directives), null);
+    records = pipeline.execute(records);
+
+    Statistics meta = new BasicStatistics();
+
+    Record summary = meta.aggregate(records);
+
+    Record stats = (Record) summary.getValue("stats");
+    Record types = (Record) summary.getValue("types");
+
+
+    System.out.println("General Statistics");
+    System.out.println("Total number of records : " + summary.getValue("total"));
+    System.out.println();
+    List<KeyValue<String, Object>> fields = stats.getFields();
+    for (KeyValue<String, Object> field : fields) {
+      List<KeyValue<String, Double>> values = (List<KeyValue<String, Double>>) field.getValue();
+      for (KeyValue<String, Double> value : values) {
+        Double percentage = value.getValue() * 100;
+        if(percentage < 20) {
+          continue;
+        }
+        System.out.println(String.format("%10s %-20s %3.2f%%", field.getKey(), value.getKey(), value.getValue() * 100));
+      }
+    }
+
+    System.out.println();
+    System.out.println("Type Statistics");
+    System.out.println();
+    fields = types.getFields();
+    for (KeyValue<String, Object> field : fields) {
+      List<KeyValue<String, Double>> values = (List<KeyValue<String, Double>>) field.getValue();
+      for (KeyValue<String, Double> value : values) {
+        Double percentage = value.getValue() * 100;
+        /*
+        if(percentage < 20) {
```

Comment:
Actually don't need this. Even if a percentage is greater than 20 I still want that to be logged. I will delete that.